src/pages/global-authorization/forms/FieldGlobalForm.js

```diff
@@ -0,0 +1,216 @@
+import { SystemRepository } from 'src/repositories/SystemRepository'
+import { AccessControlRepository } from 'src/repositories/AccessControlRepository'
+import ResourceComboBox from 'src/components/Shared/ResourceComboBox'
+import { DataSets } from 'src/resources/DataSets'
+import { CommonContext } from 'src/providers/CommonContext'
+import CustomTextField from 'src/components/Inputs/CustomTextField'
+import { AuthContext } from 'src/providers/AuthContext'
+import { getFormattedNumber } from 'src/lib/numberField-helper'
+import { DataGrid } from 'src/components/Shared/DataGrid'
+
+export default function FieldGlobalForm({ labels, maxAccess, resourceId, resourceName, setErrorMessage }) {
+  const { getRequest, postRequest } = useContext(RequestsContext)
+  const { user, setUser } = useContext(AuthContext)
+  const [isLoading, setIsLoading] = useState(false)
+
+  const invalidate = useInvalidate({
+    endpointId: SystemRepository.ModuleClassRES.qry
+  })
+
+  const detailsFormik = useFormik({
+    enableReinitialize: true,
+    validateOnChange: true,
+    initialValues: {
+      rows: [],
+      resourceId: resourceId,
+      resourceName: resourceName
+    },
+    onSubmit: async obj => {
+
+      const updatedRows = detailsFormik.values.rows
+        .filter(obj => {
+          return obj.accessLevel != null
+        }).map(({ ...rest }) => ({
+          resourceId: resourceId,
+          ...rest
+        }))
+
+      // Create the resultObject
+      const resultObject = {
+        resourceId: resourceId,
+        controls: updatedRows
+      }
+
+      const response = await postRequest({
+        extension: AccessControlRepository.GlobalControlAuthorizationPack.set2,
+        record: JSON.stringify(resultObject)
+      })
+
+      toast.success('Record Edited Successfully')
+
+      invalidate()
+    }
+  })
+
+  const columns = [
+    {
+      component: 'textfield',
+      label: labels.controlId,
+      name: 'controlId',
+      props: {
+        readOnly: true
+      }
+    },
+    {
+      component: 'textfield',
+      label: labels.controlName,
+      name: 'name',
+      props: {
+        readOnly: true
+      }
+    },
+    {
+      component: 'resourcecombobox',
+      name: 'accessLevelCombo',
+      label: labels.accessLevel,
+      props: {
+        datasetId: DataSets.AU_RESOURCE_CONTROL_ACCESS_LEVEL,
+        valueField: 'key',
+        displayField: 'value',
+        mapping: [ { from: 'key', to: 'accessLevel' } ,
+          { from: 'value', to: 'accessLevelName' }]
+      },
+      async onChange({ row: { update, newRow } }) {
+        if (!newRow?.accessLevelCombo?.key) {
+          return
+        } else {
+          update({ accessLevelName: newRow.accessLevelCombo?.value, accessLevel: newRow.accessLevelCombo?.key })
+        }
+      }
+    }
+  ]
+
+  useEffect(() => {
+    ;(async function () {
+      try {
+        if (resourceId) {
+          setIsLoading(true)
+
+          const res = await getRequest({
```

Comment:
Can we use useResourceQuery instead? Has isLoading out of the box.

Reply:
i guess here it is better to keep it query.. and what about has isloading? what do you mean?